adapter/internal/oasparser/envoyconf/http_filters.go

```diff
@@ -102,6 +113,41 @@ func getUpgradeFilters() []*hcmv3.HttpFilter {
 	return upgradeFilters
 }
 
+// getRateLimitFilter configures the ratelimit filter
+func getRateLimitFilter() *hcmv3.HttpFilter {
+	conf, _ := config.ReadConfigs()
+	rateLimit := &rate_limit.RateLimit{
+		Domain:          "default",
+		FailureModeDeny: true,
+		RateLimitService: &envoy_config_ratelimit_v3.RateLimitServiceConfig{
+			TransportApiVersion: corev3.ApiVersion_V3,
+			GrpcService: &corev3.GrpcService{
+				TargetSpecifier: &corev3.GrpcService_EnvoyGrpc_{
+					EnvoyGrpc: &corev3.GrpcService_EnvoyGrpc{
+						ClusterName: rateLimitClusterName,
+					},
+				},
+				Timeout: &durationpb.Duration{
+					Nanos: ((int32(conf.Envoy.RateLimit.RequestTimeoutInMillis) * 1000000) -
+						(int32(conf.Envoy.RateLimit.RequestTimeoutInMillis)/1000)*1000000000),
```

Comment:
Seems the 2nd line was forgotten when simplifying. We can directly multiply that too by 1000000 right?

Reply:
We need to do the integer division fisrt.

Reply:
I have missed the purpose of the line. Thanks for the explanation during the call.